types/node/globals.d.ts

```diff
@@ -689,7 +689,7 @@ declare namespace NodeJS {
     type ExitListener = (code: number) => void;
     type RejectionHandledListener = (promise: Promise<any>) => void;
     type UncaughtExceptionListener = (error: Error) => void;
-    type UnhandledRejectionListener = (reason: {} | null | undefined, promise: Promise<any>) => void;
+    type UnhandledRejectionListener = (reason: Error | {} | null | undefined, promise: Promise<any>) => void;
```

Comment:
The [document](https://nodejs.org/api/process.html#process_event_unhandledrejection) says the type is `Error | any` which translates to `any` (or `unknown`) in TypeScript.

Reply:
> Strictly speaking Error extends {} this will still require a cast and will add no value to the typings/the consumer/developer.

Yes and no. Recently I updated the typings for node in one of my projects from 10.x to 12.x and this was the first thing that didn't work. The code relied on the "reason" object to be an Error with a "stack" property. With this fix, it doesn't complain anymore and upgrading to 12.x typings would be less painful.

> The document says the type is Error | any which translates to any (or unknown) in TypeScript.

That's reasonable. It looks to me that defining this as "any" makes more sense here. What do you think?

Reply:
[that's what I said](https://github.com/DefinitelyTyped/DefinitelyTyped/pull/35906#discussion_r289683473)